rotkehlchen/globaldb/handler.py

```diff
@@ -839,46 +862,176 @@ def get_evm_tokens(
         - Protocol for which to return tokens
         - ignore_spam (default True) to filter out "spam protocol" assets
         """
+        return GlobalDBHandler._get_tokens_from_db(
+            table_name='evm_tokens',
+            token_deserializer=EvmToken.deserialize_from_db,
+            token_type_name='EvmToken',
+            exceptions=exceptions,
+            protocol=protocol,
+            ignore_spam=ignore_spam,
+            chain_filter=('chain', chain_id.serialize_for_db()),
+            needs_underlying_tokens=True,
+            extra_columns=['chain'],
+        )
+
+    @staticmethod
+    def _get_single_token_from_db(
+            table_name: str,
+            token_deserializer: Callable,
+            token_type_name: str,
+            address_filter: tuple[str, Any],
+            chain_filter: tuple[str, Any] | None = None,
+            needs_underlying_tokens: bool = False,
```

Comment:
this should be a check inside the function and decided by checking the klass that you are using. Making it an argument is a mistake because the developer can provide a wrong combination